### ISSUE TITLE
Duplicate declaration: Package[wget] is already declared

### ISSUE THREAD

**Issue by Boran** [codedellemc/puppet-scaleio#21]
HI,

I have wget already define in a base common class that is applied to all servers, so when doing "class { 'scaleio::sds_server': }" one gets:

scaleio::sds_server: Duplicate declaration: Package[wget] is already declared

Should such a common base package really be defined as a dependency? I understand why one does it, but unfortunately puppet does not allow several packages to have the same dependency.

(testing on Ubuntu 16.04 from master)

**Comment by Andrey-mp** [comment]
Hi,
The 'wget' package is needed for downloading packages for installation.
And I found that some CentOS systems don't have it by default.

It's strange that only 'wget' package raises an error - it is not only one in the common Package list.

**Comment by Boran** [comment]
Yes. I did a grep of common puppet modules I use ..
 *  puppetlabs-apt, for example uses wget but does not declare it as a dependency.
 * Sonar uses whet but includes https://github.com/maestrodev/puppet-wget which provides a puppet interfaces to wget and coudl presumably be included in several packages.

I wonder what the best practise is for avoiding dependency redeclaration errors in puppet? 
Perhaps the idea here_ http://stackoverflow.com/questions/26205727/duplicate-declaration-of-same-resource-defined-in-separate-classes#26215336 is a solution, 
or just
```if ! defined(Package['wget']) {
    package { 'wget': ensure => installed }
}```

**Comment by Andrey-mp** [comment]
Yeah, I found a similar answer - http://stackoverflow.com/questions/15266347/puppet-test-if-a-package-already-defined

**Comment by Andrey-mp** [comment]
I fix this issue in last commit.
Please recheck.

**Comment by Boran** [comment]
I pulled master, but don't see a fix.

**Comment by Andrey-mp** [comment]
https://github.com/codedellemc/puppet-scaleio/commit/c1327925e52480051a2cb2dc4a0ea414c754976b

**Comment by Boran** [comment]
Ok, yes I have that in master, but I had been looking for something like "! defined"
Where is ensure_resource defined, because it does not work as expected for me. I still get

Duplicate declaration: Package[wget] is already declared in file /mymodule/manifests/init.pp:110; cannot redeclare at /etc/puppet/environments/production/modules/scaleio/manifests/common_server.pp:12

Ah, it is in puppetlabs-stdlib. It would be useful to explain your fix :-)

So a fix of the fix: if "ensure_resource('package', ['libaio', 'numactl', 'wget'], {'ensure' => 'installed'})" is replaced with "ensure_packages(['libaio', 'numactl', 'wget'], {'ensure' => 'present'})" it works fine (for me).

**Comment by Andrey-mp** [comment]
Is all ok now? :)

**Comment by Boran** [comment]
there are no changes on master

git log -1
commit c1327925e52480051a2cb2dc4a0ea414c754976b
...
Date:   Fri Nov 18 12:17:29 2016 +0300
    use ensure_resource for common resources

**Comment by Andrey-mp** [comment]
This commit changes behavior in 'wget' definition with ensure_resource.
I meant that it should fix your error with re-definition.

**Comment by Boran** [comment]
It does not, see 4 messages higher. I had expected that you commit another fix this morning, e.g. with ensure_packages

**Comment by Andrey-mp** [comment]
hm, but can you check why ensure_resource doesn't work?
